src/main/scala/is/hail/expr/ir/MatrixValue.scala

```diff
@@ -182,10 +182,20 @@ case class MatrixValue(
       val hc = HailContext.get
       val signature = typ.colType
 
-      UnpartitionedRVD(
-        signature,
-        ContextRDD.parallelize(hc.sc, colValues.value.toArray.map(_.asInstanceOf[Row]))
-          .cmapPartitions { (ctx, it) => it.toRegionValueIterator(ctx.region, signature) })
+      if (typ.colKey.isEmpty) {
+        UnpartitionedRVD(
+          signature,
+          ContextRDD.parallelize(hc.sc, colValues.value.toArray.map(_.asInstanceOf[Row]))
+            .cmapPartitions { (ctx, it) => it.toRegionValueIterator(ctx.region, signature) })
+      } else {
+        val (_, keyF) = typ.colType.select(typ.colKey)
+        val sorted = colValues.value.map(_.asInstanceOf[Row]).sortBy(keyF)(typ.colKeyStruct.ordering.toOrdering.asInstanceOf[Ordering[Row]])
```

Comment:
@danking When you review, could you see if anything jumps out at you that might be causing the OOM error when running the doctest on the GWAS tutorial? This is the only thing I can think of that might be a problem.

Reply:
Something large is getting kryo serialized...